components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -1016,8 +1039,36 @@ void AIChatService::AssociateContent(
     return;
   }
 
-  MaybeAssociateContentWithConversation(conversation, content->GetContentId(),
-                                        content->GetWeakPtr());
+  conversation->associated_content_manager()->AddContent(content);
+
+  // Record that this is the latest conversation for this content. Even
+  // if we don't call SetAssociatedContentDelegate, the conversation still
+  // has a default Tab's navigation for which is is associated. The Conversation
+  // won't use that Tab's Page for context.
+  content_conversations_.insert_or_assign(
+      content->GetContentId(), conversation->get_conversation_uuid());
+}
+
+void AIChatService::DisassociateContent(
+    ConversationHandler::AssociatedContentDelegate* content,
+    const std::string& conversation_uuid) {
+  CHECK(content);
+
+  // Note: This will only work if the conversation is already loaded.
+  auto* conversation = GetConversation(conversation_uuid);
+  if (!conversation) {
+    return;
+  }
+
+  conversation->associated_content_manager()->RemoveContent(content);
```

Comment:
I think passing `true` here will fix the bug with the toggle still displaying after unassociating all content from a new conversation

Reply:
:facepalm: that was actually a bug - the default argument to the function should've been `true` not `false`